### PR TITLE
macOS ZFS Sandbox

### PR DESCRIPTION
**Pull request by mtelvers**
Rebase and squash of the current macOS worker.

### CODE REVIEW

File: lib/sandbox.mli [hunk 1]
```diff
@@ -12,3 +12,5 @@ val cmdliner : config Cmdliner.Term.t
 val create : state_dir:string -> config -> t Lwt.t
 (** [create ~state_dir config] is a sandboxing system that keeps state in [state_dir]
     and is configured using [config]. *)
+
+val finished : unit -> unit Lwt.t
```
Comment (tmcgilchrist) [inline lib/sandbox.mli:16]: Worth adding comment here describing what `finished` is logically doing in a sandbox?

**Review by tmcgilchrist — APPROVED** [review]
Much simpler without the rsync hacks.

File: lib/sandbox.mli (hunk 1 above)
Comment (mtelvers) [reply lib/sandbox.mli:16]: I have added a description.

**Comment by mtelvers** [comment]
> Is the GitHub actions failure a bug related to the new implementation of `finished` for macOS sandboxes?

It was from an earlier Implementation before I moved the code to the sandbox.  This line needed to be removed from the `build` function:

```
 Zfs.unmount t ~ds >>= fun () ->
```

**Pull request merged**